src/blockchain/node/HeaderOracle.cpp

```diff
@@ -438,18 +431,17 @@ auto HeaderOracle::best_hashes(
     const auto limitIsZero = (0 == limit);
     auto current{start};
     const auto tip = best_chain(lock);
-    const auto last = [&] {
-        if (limitIsZero) {
+    block::Height last{};
```

Comment:
I would speculatively initialize it to tip.first and change it if(!limitZero), without an else clause.